Release notes for RC.2 (#2585)
dapr/dapr: docs/release_notes/v1.0.0-rc.2.md

```bash
dapr init --runtime-version=1.0.0-rc.2
```

Wait for the update to finish,  ensure you are using the latest version of Dapr(1.0.0-rc.2) with:

```bash
$ dapr --version

CLI version: 1.0.0-rc.3
Runtime version: 1.0.0-rc.2
```

### Kubernetes

#### Upgrading from previous version

If you previously installed Dapr using Helm, you can upgrade Dapr to a new version.
If you installed Dapr using the CLI, go [here](#starting-fresh-install-on-a-cluster).

##### 1. Get the latest CLI

Download the latest RC release from [here](https://github.com/dapr/cli/releases) and put the `dapr` binary in your PATH.

##### 2. Upgrade from existing cluster running 0.11.x

Follow instructions below to migrate from 0.11.x placement service to 1.0.0-rc.2 placement service safely.

First run these two commands to disable `helm upgrade` from uninstall 0.11.x placement service.

```sh
kubectl annotate deployment dapr-placement helm.sh/resource-policy=keep -n dapr-system
kubectl annotate svc dapr-placement helm.sh/resource-policy=keep -n dapr-system
```

Then, export certs:

```sh
dapr mtls export -o ./certs
```

Now, upgrade Dapr to 1.0.0-rc.2 using the below commands:

```sh
helm repo update

> helm upgrade dapr dapr/dapr --version 1.0.0-rc.2 --namespace dapr-system --reset-values --set-file dapr_sentry.tls.root.certPEM=./certs/ca.crt --set-file dapr_sentry.tls.issuer.certPEM=./certs/issuer.crt --set-file dapr_sentry.tls.issuer.keyPEM=./certs/issuer.key --set global.ha.enabled=true --wait

Reviewer: ```suggestion
helm upgrade dapr dapr/dapr --version 1.0.0-rc.2 --namespace dapr-system --reset-values --set-file dapr_sentry.tls.root.certPEM=./certs/ca.crt --set-file dapr_sentry.tls.issuer.certPEM=./certs/issuer.crt --set-file dapr_sentry.tls.issuer.keyPEM=./certs/issuer.key --set global.ha.enabled=true --wait

# Due to Helm being unable to upgrade CRDs, we will need to upgrade CRD directly.
kubectl apply -f https://raw.githubusercontent.com/dapr/dapr/release-1.0/charts/dapr/crds/configuration.yaml
kubectl apply -f https://raw.githubusercontent.com/dapr/dapr/release-1.0/charts/dapr/crds/components.yaml
```

##### 1. Get the latest CLI

Download the latest RC release from [here](https://github.com/dapr/cli/releases) and put the `dapr` binary in your PATH.

##### 2. Upgrade from existing cluster running 0.11.x

Follow instructions below to migrate from 0.11.x placement service to 1.0.0-rc.2 placement service safely.

First run these two commands to disable `helm upgrade` from uninstall 0.11.x placement service.

```sh
kubectl annotate deployment dapr-placement helm.sh/resource-policy=keep -n dapr-system
kubectl annotate svc dapr-placement helm.sh/resource-policy=keep -n dapr-system
```

Then, export certs:

```sh
dapr mtls export -o ./certs
```

Now, upgrade Dapr to 1.0.0-rc.2 using the below commands:

```sh
helm repo update
helm upgrade dapr dapr/dapr --version 1.0.0-rc.2 --namespace dapr-system --reset-values --set-file dapr_sentry.tls.root.certPEM=./certs/ca.crt --set-file dapr_sentry.tls.issuer.certPEM=./certs/issuer.crt --set-file dapr_sentry.tls.issuer.keyPEM=./certs/issuer.key --set global.ha.enabled=true --wait
```

Due to Helm being unable to upgrade CRDs, we will need to upgrade CRD directly.
```sh

> kubectl apply -f https://raw.githubusercontent.com/dapr/dapr/21636a9237f2dcecd9c80f329e99b512e8377739/charts/dapr/crds/configuration.yaml

Reviewer: ```suggestion
kubectl replace -f https://raw.githubusercontent.com/dapr/dapr/21636a9237f2dcecd9c80f329e99b512e8377739/charts/dapr/crds/configuration.yaml
```

Reviewer: This is the only place that needs `kubectl replace` out of the 4.

dapr-sentry-84565c747b-fdlls             1/1     Running   0          41s
dapr-sentry-84565c747b-ldnsf             1/1     Running   0          29s
dapr-sidecar-injector-68f868668f-6xnbt   1/1     Running   0          41s
dapr-sidecar-injector-68f868668f-j7jcq   1/1     Running   0          29s
dapr-sidecar-injector-68f868668f-ltxq4   1/1     Running   0          36s
```

Restart your application deployments to update the Dapr runtime.
```sh
kubectl rollout restart deploy/<DEPLOYMENT-NAME>
```

Once the deployment is completed, delete the 0.11.x dapr-placement service by running the following commands:
```sh
kubectl delete deployment dapr-placement -n dapr-system
kubectl delete svc dapr-placement -n dapr-system
```

All done!

##### 2. Upgrade from existing cluster running 1.0.0-rc.1

Follow instructions below to migrate from 0.11.x placement service to 1.0.0-rc.2 placement service safely.

First run these two commands to disable `helm upgrade` from uninstall 0.11.x placement service.

Then, export certs:

```sh
dapr mtls export -o ./certs
```

Now, upgrade Dapr to 1.0.0-rc.2 using the below commands:

```sh
helm repo update

> helm upgrade dapr dapr/dapr --version 1.0.0-rc.2 --namespace dapr-system --reset-values --set-file dapr_sentry.tls.root.certPEM=./certs/ca.crt --set-file dapr_sentry.tls.issuer.certPEM=./certs/issuer.crt --set-file dapr_sentry.tls.issuer.keyPEM=./certs/issuer.key --set global.ha.enabled=true --wait

Reviewer: ```suggestion
helm upgrade dapr dapr/dapr --version 1.0.0-rc.2 --namespace dapr-system --reset-values --set-file dapr_sentry.tls.root.certPEM=./certs/ca.crt --set-file dapr_sentry.tls.issuer.certPEM=./certs/issuer.crt --set-file dapr_sentry.tls.issuer.keyPEM=./certs/issuer.key --set global.ha.enabled=true --wait

```suggestion
helm upgrade dapr dapr/dapr --version 1.0.0-rc.2 --namespace dapr-system --reset-values --set-file dapr_sentry.tls.root.certPEM=./certs/ca.crt --set-file dapr_sentry.tls.issuer.certPEM=./certs/issuer.crt --set-file dapr_sentry.tls.issuer.keyPEM=./certs/issuer.key --set global.ha.enabled=true --wait

# Due to Helm being unable to upgrade CRDs, we will need to upgrade CRD directly.
kubectl apply -f https://raw.githubusercontent.com/dapr/dapr/release-1.0/charts/dapr/crds/configuration.yaml
kubectl apply -f https://raw.githubusercontent.com/dapr/dapr/release-1.0/charts/dapr/crds/components.yaml
```
```

- **FIX** Bug in flattening of JSON hierarchy by local file secrets store. [550](https://github.com/dapr/components-contrib/issues/550)

### Dashboard

- **FIX** Display dashboard version correctly in command line interface [86](https://github.com/dapr/dashboard/issues/86)

- **FIX** Display Placement service in Dashboard list of Dapr services [101](https://github.com/dapr/dashboard/issues/101)

- **FIX** Display dashboard in list of Dapr services [102](https://github.com/dapr/dashboard/issues/102)

- **ADD** Metrics setting in Configurations screen [103](https://github.com/dapr/dashboard/issues/103)

- UPDATE Configuration to auto-fill screen [105](https://github.com/dapr/dashboard/issues/105)

- **UPDATE** Component configuration to auto-fill screen [106](https://github.com/dapr/dashboard/issues/106)

- UPDATE Application configuration to auto-fill screen [107](https://github.com/dapr/dashboard/issues/107)

- **FIX** display of multiple replicas of Dapr services [108](https://github.com/dapr/dashboard/issues/108)

- **UPDATE** Application list action from trash can icon to text button [109](https://github.com/dapr/dashboard/issues/109)

### .NET SDK

> - **RESOLVED** Dependency injection for Actors [171](https://github.com/dapr/dotnet-sdk/issues/171)

Reviewer 2: ```suggestion
- **RESOLVED** Dependency injection for Actors is enabled by default for ASP.NET Core projects  [171](https://github.com/dapr/dotnet-sdk/issues/171)```

- **RESOLVED** Dapr Actors should not use BinaryFormatter [414](https://github.com/dapr/dotnet-sdk/issues/414)

> - **RESOLVED** HTTPExtension feedback [421](https://github.com/dapr/dotnet-sdk/issues/421)

Reviewer 2: ```suggestion
- **Update** Naming of HTTPExtension and related types to follow .NET conventions [421](https://github.com/dapr/dotnet-sdk/issues/421)
```

Once you have installed the CLI, run:

```bash
dapr init --runtime-version=1.0.0-rc.2
```

Wait for the update to finish,  ensure you are using the latest version of Dapr(1.0.0-rc.2) with:

```bash
$ dapr --version

CLI version: 1.0.0-rc.3
Runtime version: 1.0.0-rc.2
```

### Kubernetes

#### Upgrading from previous version

If you previously installed Dapr using Helm, you can upgrade Dapr to a new version.
If you installed Dapr using the CLI, go [here](#starting-fresh-install-on-a-cluster).

##### 1. Get the latest CLI

Download the latest RC release from [here](https://github.com/dapr/cli/releases) and put the `dapr` binary in your PATH.

##### 2. Upgrade from existing cluster running 0.11.x

Follow instructions below to migrate from 0.11.x placement service to 1.0.0-rc.2 placement service safely.

First run these two commands to disable `helm upgrade` from uninstall 0.11.x placement service.

```sh
kubectl annotate deployment dapr-placement helm.sh/resource-policy=keep -n dapr-system
kubectl annotate svc dapr-placement helm.sh/resource-policy=keep -n dapr-system
```

> 

Reviewer: ```suggestion

Now, check if you have any exporter type components, by running the following command `kubectl get components -o=custom-columns='NAME:metadata.name,TYPE:spec.type'` and look for components with type `exporter.*`.

Since exporter components are no longer supported, you will need to remove them prior to upgrading. You may want to print their content for your reference using `kubectl get component <component-name> -o yaml`.

```

Then, export certs:

```sh
dapr mtls export -o ./certs
```

Now, upgrade Dapr to 1.0.0-rc.2 using the below commands:

```sh
helm repo update
helm upgrade dapr dapr/dapr --version 1.0.0-rc.2 --namespace dapr-system --reset-values --set-file dapr_sentry.tls.root.certPEM=./certs/ca.crt --set-file dapr_sentry.tls.issuer.certPEM=./certs/issuer.crt --set-file dapr_sentry.tls.issuer.keyPEM=./certs/issuer.key --set global.ha.enabled=true --wait
```

Due to Helm being unable to upgrade CRDs, we will need to upgrade CRD directly.
```sh
kubectl replace -f https://raw.githubusercontent.com/dapr/dapr/21636a9237f2dcecd9c80f329e99b512e8377739/charts/dapr/crds/configuration.yaml
kubectl replace -f https://raw.githubusercontent.com/dapr/dapr/21636a9237f2dcecd9c80f329e99b512e8377739/charts/dapr/crds/components.yaml
```

> 

Reviewer: ```suggestion

To re-incorporate your tracing settings, follow the [observability docs](https://docs.dapr.io/operations/monitoring/) to leverage the new trace settings in the Configuration resource.

```

dapr-placement-server-0                  1/1     Running   0          41s
dapr-placement-server-1                  1/1     Running   0          41s
dapr-placement-server-2                  1/1     Running   0          41s
dapr-sentry-84565c747b-7bh8h             1/1     Running   0          35s
dapr-sentry-84565c747b-fdlls             1/1     Running   0          41s
dapr-sentry-84565c747b-ldnsf             1/1     Running   0          29s
dapr-sidecar-injector-68f868668f-6xnbt   1/1     Running   0          41s
dapr-sidecar-injector-68f868668f-j7jcq   1/1     Running   0          29s
dapr-sidecar-injector-68f868668f-ltxq4   1/1     Running   0          36s
```

Restart your application deployments to update the Dapr runtime.
```sh
kubectl rollout restart deploy/<DEPLOYMENT-NAME>
```

Once the deployment is completed, delete the 0.11.x dapr-placement service by running the following commands:
```sh
kubectl delete deployment dapr-placement -n dapr-system
kubectl delete svc dapr-placement -n dapr-system
```

All done!

##### 2. Upgrade from existing cluster running 1.0.0-rc.1

First, export certs:

```sh
dapr mtls export -o ./certs
```

> 

Reviewer: ```suggestion

Now, check if you have any exporter type components, by running the following command `kubectl get components -o=custom-columns='NAME:metadata.name,TYPE:spec.type'` and look for components with type `exporter.*`.

Since exporter components are no longer supported, you will need to remove them prior to upgrading. You may want to print their content for your reference using `kubectl get component <component-name> -o yaml`.

```

Now, upgrade Dapr to 1.0.0-rc.2 using the below commands:

```sh
helm repo update
helm upgrade dapr dapr/dapr --version 1.0.0-rc.2 --namespace dapr-system --reset-values --set-file dapr_sentry.tls.root.certPEM=./certs/ca.crt --set-file dapr_sentry.tls.issuer.certPEM=./certs/issuer.crt --set-file dapr_sentry.tls.issuer.keyPEM=./certs/issuer.key --set global.ha.enabled=true --wait
```

Due to Helm being unable to upgrade CRDs, we will need to upgrade CRD directly.
```sh
kubectl replace -f https://raw.githubusercontent.com/dapr/dapr/21636a9237f2dcecd9c80f329e99b512e8377739/charts/dapr/crds/configuration.yaml
kubectl replace -f https://raw.githubusercontent.com/dapr/dapr/21636a9237f2dcecd9c80f329e99b512e8377739/charts/dapr/crds/components.yaml
```

> 

Reviewer: ```suggestion

To re-incorporate your tracing settings, follow the [observability docs](https://docs.dapr.io/operations/monitoring/) to leverage the new trace settings in the Configuration resource.

```

dapr-placement-server-0                 1/1     Running   0          13m
dapr-sentry-5974c7cb6d-9bnqd            1/1     Running   0          13m
dapr-sidecar-injector-5b5bb9454-sfmmx   1/1     Running   0          13m
```

After Dapr 1.0.0-rc.2 has been installed, perform a rolling restart for your deployments to pick up the new version of the sidecar.
This can be done with:

```
kubectl rollout restart deploy/<deployment-name>
```

## Breaking Changes

### Dapr Runtime

- **FIX** Pubsub API will now return 403 for forbidden pubsub topics. [2323](https://github.com/dapr/dapr/issues/2323)

### Dapr CLI

- **MAKE** shorthands across commands consistent [530](https://github.com/dapr/cli/issues/530)

- **UPDATE** Default standalone Configuration to not use deprecated Zipkin exporter component. [534](https://github.com/dapr/cli/issues/534)

- **ADD** --publish-app-id to the publish command [536](https://github.com/dapr/cli/issues/536)

- **REMOVED** unused and unsupported `--image` flag from `run` command. [537](https://github.com/dapr/cli/issues/537)

- **REMOVED** `redis-host` flag from `dapr init` [540](https://github.com/dapr/cli/issues/540)

### .NET SDK

> - **RESOLVED** Dependency injection for Actors [171](https://github.com/dapr/dotnet-sdk/issues/171)

Reviewer 2: ```suggestion
- **RESOLVED** Dependency injection for Actors is enabled by default for ASP.NET Core projects  [171](https://github.com/dapr/dotnet-sdk/issues/171)
```

- **RESOLVED** Dapr Actors should not use BinaryFormatter [414](https://github.com/dapr/dotnet-sdk/issues/414)

> - **RESOLVED** HTTPExtension feedback [421](https://github.com/dapr/dotnet-sdk/issues/421)

Reviewer 2: ```suggestion
- **Update** Naming of HTTPExtension and related types to follow .NET conventions [421](https://github.com/dapr/dotnet-sdk/issues/421)
```

*Note: Make sure you're uninstalling with your existing CLI version*

```bash
dapr uninstall --kubernetes
```

If you previously installed Dapr using __Helm 3.X__:

```bash
helm uninstall dapr -n dapr-system
```

If you installed Dapr with Helm to a namespace other than `dapr-system`, modify the uninstall command above to account for that.

##### Post installation

Verify the control plane pods are running and are healthy:

```
$ kubectl get pods -n dapr-system
NAME                                    READY   STATUS    RESTARTS   AGE
dapr-dashboard-85cc6c9f7d-hzk69         1/1     Running   0          13m
dapr-operator-84985cb985-h2zdv          1/1     Running   0          13m
dapr-placement-server-0                 1/1     Running   0          13m
dapr-sentry-5974c7cb6d-9bnqd            1/1     Running   0          13m
dapr-sidecar-injector-5b5bb9454-sfmmx   1/1     Running   0          13m
```

After Dapr 1.0.0-rc.2 has been installed, perform a rolling restart for your deployments to pick up the new version of the sidecar.
This can be done with:

```
kubectl rollout restart deploy/<deployment-name>
```

## Breaking Changes

> ### Dapr Runtime

Reviewer 3: It'd be good to have the mitigation @tcnghia wrote for the highlight section about components here in the breaking changes. If you're looking for an issue reference this seems like a good top level and is already referenced up there:

https://github.com/dapr/dapr/issues/2337

Author: We have that on the top, not? This is just a simple list of issues.

Reviewer 3: Similar to above, I'd like to keep it simple so all I'm asking for is something at the end of the list like:
Note: See #2337 for more details.

Author: This feels more like a copy and paste from the existing upgrade instructions.

- **FIXED** Pubsub API will now return 403 for forbidden pubsub topics. [2323](https://github.com/dapr/dapr/issues/2323)

- **MOVED** Zipkin trace configuration from Zipkin exporter into Dapr core [2521](https://github.com/dapr/dapr/issues/2521)

### Dapr CLI

- **MADE** shorthands across commands consistent [530](https://github.com/dapr/cli/issues/530)

- **UPDATED** Default standalone Configuration to not use deprecated Zipkin exporter component. [534](https://github.com/dapr/cli/issues/534)

- **ADDED** `--publish-app-id` to the publish command [536](https://github.com/dapr/cli/issues/536)

- **REMOVED** unused and unsupported `--image` flag from `run` command. [537](https://github.com/dapr/cli/issues/537)

- **REMOVED** `redis-host` flag from `dapr init` [540](https://github.com/dapr/cli/issues/540)

> ### .NET SDK

Reviewer 3: It'd be good to include a link to the mitigation guide here:

https://github.com/dapr/dotnet-sdk/issues/505

Author: We do that on the top. This is just a simple list of issues closed.

Reviewer 3: Yes, and I'd like to keep it simple too. From discussions with @msfussell he feels it'd be good to reference the mitigations here so people have to spend less time digging. 

So, we don't need the whole mitigation here but could do something simple like:

Note: See [upgrade] for mitigation instructions.

Author: I can add this.

- **RESOLVED** Dependency injection for Actors is enabled by default for ASP.NET Core projects [171](https://github.com/dapr/dotnet-sdk/issues/171)

- **RESOLVED** Dapr Actors should not use BinaryFormatter [414](https://github.com/dapr/dotnet-sdk/issues/414)

- **UPDATED** Naming of HTTPExtension and related types to follow .NET conventions [421](https://github.com/dapr/dotnet-sdk/issues/421)

- **FIXED** Timer invocation after Actor runtime app restarts. [435](https://github.com/dapr/dotnet-sdk/issues/435)

- **REMOVED** public constructor on ActorTypeInfo - this is a breaking change but unlikely to impact users as it was not useful before [461](https://github.com/dapr/dotnet-sdk/issues/461)

### Go SDK

- **RESOLVED** Have consistent method/parameter names across SDKs [117](https://github.com/dapr/go-sdk/issues/117)

> ### Java SDK

Reviewer 3: Similar to dotnet sdk, it'd be good to include a link to the upgrade guide here.

https://github.com/dapr/java-sdk/issues/415

Author: It is in the top, this is just a simple list of issues.

Reviewer 3: Similar to above, I'd like to keep it simple so all I'm asking for is something at the end of the list like:
Note: See https://github.com/dapr/java-sdk/issues/415 for more details.

Author: Done.

  
# Dapr 1.0.0-rc.2

We're happy to announce the release of Dapr 1.0.0-rc.2!

We would like to extend our [thanks to all](#Acknowledgements) new and existing contributors who helped make this release happen.

This is a release candidate version as we progress towards Dapr v1.0.0. We appreciate you trying Dapr and providing feedback on this release.

See [this](#upgrading-to-dapr-100-rc2) section on upgrading Dapr to version 1.0.0-rc.2.

> Dapr CLI version in this release is 1.0.0-rc.3 and is not compatible with runtime versions prior to 1.0.0-rc.2. Download Dapr CLI 1.0.0-rc.3 from [here](https://github.com/dapr/cli/releases) and put the `dapr` binary in your PATH, or use the rc1 install script in the Getting Started guide.

> For Java applications using Dapr Java SDK, use SDK version 1.0.0-rc-2. See [upgrade instructions for the Java SDK](https://github.com/dapr/java-sdk/issues/415).

> For .Net applications using Dapr .Net SDK, use SDK version 1.0.0-rc02. See [upgrade instructions for the .Net SDK](https://github.com/dapr/dotnet-sdk/issues/505).

> **Note: This release contains a few [breaking changes](#breaking-changes).**

**Highlights**

> Actors in Java SDK communicate to Dapr sidecar via GRPC.
> 

Reviewer: ```suggestion
* Actors in Java SDK communicate to Dapr sidecar via GRPC.
* Deprecation of exporters component [#2337](https://github.com/dapr/dapr/issues/2337):  Tracing is now part of Dapr core, and tracing settings have moved to Configuration CRD. See [this doc](https://v1-rc2.docs.dapr.io/operations/configuration/configuration-overview/#tracing) for information about the new tracing settings. For more detailed instructions on setting up tracing in new Configuration, check out the [observability quickstart sample](https://github.com/dapr/quickstarts/tree/v1.0.0-rc.2/observability) as well as backend-specific guides for [Zipkin](https://v1-rc2.docs.dapr.io/operations/monitoring/zipkin/), [Jaeger](https://v1-rc2.docs.dapr.io/operations/monitoring/jaeger/), [New Relics](https://v1-rc2.docs.dapr.io/operations/monitoring/newrelic/), and [OpenTelemetry trace backends like Azure AppInsight](https://v1-rc2.docs.dapr.io/operations/monitoring/open-telemetry-collector/).
```

## New in this release

### Dapr Runtime

> - **RESOLVED** [E2E Scenario Tests] for Actor Service Invocation  [1759](https://github.com/dapr/dapr/issues/1759)

Author: We should not. I will remove those.

- **RESOLVED** Azure Key Vault Secret Store can read different secret versions. [522](https://github.com/dapr/components-contrib/issues/522)

- **RESOLVED** Getting a weird error when trying to log more [529](https://github.com/dapr/components-contrib/issues/529)

- **ADD** Postmark output binding [545](https://github.com/dapr/components-contrib/issues/545)

- **RESOLVED** Add access level to Azure Blob Storage [548](https://github.com/dapr/components-contrib/issues/548)

- **FIX** Bug in flattening of JSON hierarchy by local file secrets store. [550](https://github.com/dapr/components-contrib/issues/550)

### Dashboard

- **FIX** Display dashboard version correctly in command line interface [86](https://github.com/dapr/dashboard/issues/86)

- **FIX** Display Placement service in Dashboard list of Dapr services [101](https://github.com/dapr/dashboard/issues/101)

- **FIX** Display dashboard in list of Dapr services [102](https://github.com/dapr/dashboard/issues/102)

- **ADD** Metrics setting in Configurations screen [103](https://github.com/dapr/dashboard/issues/103)

> - **UPDATE** Configuration to auto-fill screen [105](https://github.com/dapr/dashboard/issues/105)

Author: Not really, they are slightly different.

> - **UPDATE** Configuration to auto-fill screen [105](https://github.com/dapr/dashboard/issues/105)

Author: I've changed to put them all in the same line.

- **UPDATE** Component configuration to auto-fill screen [106](https://github.com/dapr/dashboard/issues/106)

- **UPDATE** Application configuration to auto-fill screen [107](https://github.com/dapr/dashboard/issues/107)

- **FIX** display of multiple replicas of Dapr services [108](https://github.com/dapr/dashboard/issues/108)

- **UPDATE** Application list action from trash can icon to text button [109](https://github.com/dapr/dashboard/issues/109)

### .NET SDK

- **RESOLVED** Dependency injection for Actors is enabled by default for ASP.NET Core projects [171](https://github.com/dapr/dotnet-sdk/issues/171)

- **RESOLVED** Dapr Actors should not use BinaryFormatter [414](https://github.com/dapr/dotnet-sdk/issues/414)

- **UPDATE** Naming of HTTPExtension and related types to follow .NET conventions [421](https://github.com/dapr/dotnet-sdk/issues/421)

- **RESOLVED** InvokeMethodAsync reports cancellation using RpcException instead of OperationCanceledOperation [422](https://github.com/dapr/dotnet-sdk/issues/422)

- **FIX** Timer invocation after Actor runtime app restarts. [435](https://github.com/dapr/dotnet-sdk/issues/435)

> - **REMOVED** public constructor on ActorTypeInfo - this is a breaking change but unlikely to impact users as it was not useful before [461](https://github.com/dapr/dotnet-sdk/issues/461)

Author: It was asked to add this here. So, we have two different ideas here. I am OK to see this called out in both. It makes it easier for people reading the release note to overlook the breaking change part.

- **UPDATE** Have consistent method/parameter names across SDKs [117](https://github.com/dapr/go-sdk/issues/117)

- **ADD** Parameterize request method verb on service invocation [118](https://github.com/dapr/go-sdk/issues/118)

- **ADD** Support HTTP query string on method invoke [125](https://github.com/dapr/go-sdk/issues/125)

### Java SDK

- **USE** DaprException consistently across the SDK. [203](https://github.com/dapr/java-sdk/issues/203)

- **UPDATE** Opentelemetry library version to 0.10.0 [379](https://github.com/dapr/java-sdk/issues/379)

- **ADD** Support for Actor APIs over GRPC from app into runtime [393](https://github.com/dapr/java-sdk/issues/393)

- **UPDATE** Java SDK trace sample at https://github.com/dapr/java-sdk/blob/v0.10.0/examples/src/main/java/io/dapr/examples/tracing/README.md to not use exporter components. [404](https://github.com/dapr/java-sdk/issues/404)

### Python SDK

- **FIX** Timer invocation after Actor runtime app restarts. [134](https://github.com/dapr/python-sdk/issues/134)

- **RESOLVED** Broken kubernetes link in examples readme [147](https://github.com/dapr/python-sdk/issues/147)

> - **UPDATE** versioning for Python SDK to 1.0 RC [151](https://github.com/dapr/python-sdk/issues/151)

Author: Done.

helm install dapr dapr/dapr --version 1.0.0-rc.2 --namespace dapr-system
```

Alternatively, you can use the latest version of CLI:

```
dapr init --runtime-version=1.0.0-rc.2 --kubernetes
```

If you previously installed Dapr on your Kubernetes cluster using the Dapr CLI, run:

*Note: Make sure you're uninstalling with your existing CLI version*

```bash
dapr uninstall --kubernetes
```

If you previously installed Dapr using __Helm 3.X__:

```bash
helm uninstall dapr -n dapr-system
```

If you installed Dapr with Helm to a namespace other than `dapr-system`, modify the uninstall command above to account for that.

##### Post installation

Verify the control plane pods are running and are healthy:

```
$ kubectl get pods -n dapr-system
NAME                                    READY   STATUS    RESTARTS   AGE
dapr-dashboard-85cc6c9f7d-hzk69         1/1     Running   0          13m
dapr-operator-84985cb985-h2zdv          1/1     Running   0          13m
dapr-placement-server-0                 1/1     Running   0          13m
dapr-sentry-5974c7cb6d-9bnqd            1/1     Running   0          13m
dapr-sidecar-injector-5b5bb9454-sfmmx   1/1     Running   0          13m
```

> After Dapr 1.0.0-rc.2 has been installed, perform a rolling restart for your deployments to pick up the new version of the sidecar.

Reviewer 4: This post installation is nested within starting a new cluster. Does the rollout command make sense here?

Author: Makes sense. Removed the rolling restart instruction.

This can be done with:

```
kubectl rollout restart deploy/<deployment-name>
```

## Breaking Changes

### Dapr Runtime

> 

Reviewer 4: exporters are removed. Zipkin is moved to configuration. 
Both here and in components-contrib.

Author: There is no issue for zipkin in components contrib in the release notes. Do you have a link to an issue?

Reviewer 4: yes there is no issue but there is a PR for the removal of the exporters. https://github.com/dapr/components-contrib/pull/553

Author: We don't link PRs unless they don't have a corresponding issue.

Due to Helm being unable to upgrade CRDs, we will need to upgrade CRD directly.
```sh
kubectl replace -f https://raw.githubusercontent.com/dapr/dapr/21636a9237f2dcecd9c80f329e99b512e8377739/charts/dapr/crds/configuration.yaml
kubectl replace -f https://raw.githubusercontent.com/dapr/dapr/21636a9237f2dcecd9c80f329e99b512e8377739/charts/dapr/crds/components.yaml
```

> Once Dapr is installed completely, ensure that 0.11.x dapr-placement service is still running and **wait until all PODs are running**

Reviewer 5: lowercase pods

Due to Helm being unable to upgrade CRDs, we will need to upgrade CRD directly.
```sh
kubectl replace -f https://raw.githubusercontent.com/dapr/dapr/21636a9237f2dcecd9c80f329e99b512e8377739/charts/dapr/crds/configuration.yaml
kubectl replace -f https://raw.githubusercontent.com/dapr/dapr/21636a9237f2dcecd9c80f329e99b512e8377739/charts/dapr/crds/components.yaml
```

> Once Dapr is installed completely, ensure until **all PODs are running**

Reviewer 5: lowercase pods
